### ISSUE TITLE
Request message requires a payload

### ISSUE THREAD

**Issue by FromZeus** [qbeon/webwire-go#22]
Request message requires a payload even though a payload shouldn't be obligatory. In some cases a request name is sufficient. For example if we just want to notify the server about an event like `client.Request("event_a_happend", wwr.Payload{})` then we won't need a payload, but currently we have to provide a payload otherwise an error is returned indicating that the request message couldn't be parsed.

```
WEBWIRE_ERR: 2018/05/15 18:26:36 serverHttp.go:72: Failed parsing message: Invalid request message, too short
```

**Comment by romshark** [comment]
I think we indeed shouldn't require a request message to include a payload though in this case we must require it to have a name:

| Name        | Payload           | Error? |
| :---: | :---: | :---: |
| ✓ | ✓ | no |
| - | ✓ | no |
| ✓ | - | no |
| - | - | **yes** |

**Comment by romshark** [comment]
As of [commit b498cca](https://github.com/qbeon/webwire-go/commit/b498cca0582bc8f197838931561daaa5969609f6) the webwire server will now accept request messages that either provide a name only or a payload only, or both. It'll reject request messages without both the name and the payload returning a `MsgInternalError` reply (can be improved by returning a new type of special reply message "MsgReplyProtoError" that's not yet implemented in the protocol version 1.4)

The server now also won't break the connection in case of a protocol/parser error.
